packages/@aws-cdk/aws-ec2/lib/vpc.ts

```diff
@@ -468,6 +481,11 @@ export class VpcSubnet extends VpcSubnetRef {
      */
     public readonly subnetId: VpcSubnetId;
 
+    /**
+     * The tags for this subnet
+     */
+    private readonly tags: cdk.Token[] = [];
```

Comment:
There's no need to store these as `Token`s internally. Just store a hash and initialize it like this:

```ts
this.tags = {
  Name: this.path,
  ...(this.props.tags || {}) // if props.tags contains "Name" it will be overridden by the splat
};
```

Then use a token to "lazy render" to CloudFormation:

```ts
const subnet = new cloudformation.SubnetResource(this, 'Subnet', {
    // ...
    tags: new Token(() => Object.keys(this.tags).map(key => ({ key, value: this.tags[key] })))
};
```

(you can also create a helper function if you think it's more readable)